torch/utils/weak.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, key, callback=None):
         # time the user attempts to hash the weakref, we can eagerly
         # cache the id of the key as we know this is definitely the hash
         # method
-        self._id = id(key)
+        self._id = id(ref(key))
```

Comment:
I actually don't really understand this. In particular, is the comment above still accurate?